driver-core/src/main/com/mongodb/client/model/bulk/ClientWriteModel.java

```diff
@@ -0,0 +1,355 @@
+     *
+     * @param namespace The namespace.
+     * @param filter The filter.
+     * @param updatePipeline The update pipeline.
+     * @return The requested model.
+     * @see Filters
+     * @see Updates
+     */
+    static ClientWriteModel updateMany(
+            final MongoNamespace namespace,
+            final Bson filter,
+            final Iterable<? extends Bson> updatePipeline) {
+        notNull("namespace", namespace);
+        notNull("filter", filter);
+        notNull("updatePipeline", updatePipeline);
+        return new ClientUpdateManyModel(namespace, filter, null, updatePipeline, null);
+    }
+
+    /**
+     * Creates a model for updating all documents in the {@code namespace} that match the {@code filter}.
+     *
+     * @param namespace The namespace.
+     * @param filter The filter.
+     * @param updatePipeline The update pipeline.
+     * @param options The options.
+     * @return The requested model.
+     * @see Filters
+     * @see Updates
+     */
+    static ClientWriteModel updateMany(
+            final MongoNamespace namespace,
+            final Bson filter,
+            final Iterable<? extends Bson> updatePipeline,
+            final ClientUpdateOptions options) {
+        notNull("namespace", namespace);
+        notNull("filter", filter);
+        notNull("updatePipeline", updatePipeline);
+        notNull("options", options);
+        return new ClientUpdateManyModel(namespace, filter, null, updatePipeline, options);
+    }
+
+    /**
+     * Creates a model for replacing at most one document in the {@code namespace} that matches the {@code filter}.
+     * This method is functionally equivalent to {@link #replaceOne(MongoNamespace, Bson, Object, ClientReplaceOptions)}
+     * with the {@linkplain ClientReplaceOptions#clientReplaceOptions() default options}.
+     *
+     * @param namespace The namespace.
+     * @param filter The filter.
+     * @param replacement The replacement.
+     * @return The requested model.
+     * @param <TDocument> The document type, for example {@link Document}.
+     * @see Filters
+     */
+    static <TDocument> ClientWriteModel replaceOne(
+            final MongoNamespace namespace,
+            final Bson filter,
+            final TDocument replacement) {
+        notNull("namespace", namespace);
+        notNull("filter", filter);
+        notNull("replacement", replacement);
+        return new ClientReplaceOneModel(namespace, filter, replacement, null);
+    }
+
+    /**
+     * Creates a model for replacing at most one document in the {@code namespace} that matches the {@code filter}.
+     *
+     * @param namespace The namespace.
+     * @param filter The filter.
+     * @param replacement The replacement.
+     * @param options The options.
+     * @return The requested model.
+     * @param <TDocument> The document type, for example {@link Document}.
+     * @see Filters
+     */
+    static <TDocument> ClientWriteModel replaceOne(
```

Comment:
`BULK-TODO` Document that `replacement` must not have field names starting with `$`, except when [DBRefs](https://www.mongodb.com/docs/manual/reference/database-references/#dbrefs) are used.